src/README.md

```diff
@@ -0,0 +1,281 @@
+            let z = &mut x;
+            world(z); // return mutably borrowed source from z to x since z is no longer used
+        }
+        fn world(s : &mut String) { 
+            s.push_str(", world")
+        }
+        ```
+    - MutableDie <a name="MutableDie"></a>
+    The MutableDie event represent return of a mutably borrowed source.
+        ```
+        MutableDie {
+            from: Option<ResourceAccessPoint>,
+            to: Option<ResourceAccessPoint>,
+        },
+        ```
+        User case:
+        ```
+        fn main() {
+            let y = &x
+            let z = &x;
+            f(y, z); // return immutably borrowed source from z to x since z is no longer used
+            // also return immutably borrowed source from y to x since y is no longer used
+        }
+        fn f(s1 : &String, s2 : &String) { 
+            println!("{} and {}", s1, s2)
+        }
+        ```
+    - PassByStaticReference <a name="PassByStaticReference"></a>
+    The PassByStaticReference event represent passing an inmutable reference to a function.
+        ```
+        PassByStaticReference {
+            from: Option<ResourceAccessPoint>,
+            to: Option<ResourceAccessPoint>, // must be a function
+        },
+        ```
+        User case:
+        ```
+        fn main() {
+            let x = String::from("hello"); 
+            f(&x); // f() could only read from x
+        }
+        fn f(s : &String) { 
+            println!("{}", s) 
+        } 
+        ```
+    - PassByMutableReference <a name="PassByMutableReference"></a>
+    The PassByMutableReference event represent passing a mutable reference to a function.
+        ```
+        PassByMutableReference {
+            from: Option<ResourceAccessPoint>,
+            to: Option<ResourceAccessPoint>, // must be a function
+        },
+        ```
+        User case:
+        ```
+        fn main() {
+            let z = &mut x;
+            world(z); // world() could read from/write to z
+        }
+        fn world(s : &mut String) { 
+            s.push_str(", world")
+        }
+        ```
+    - GoOutOfScope <a name="GoOutOfScope"></a>
+    The GoOutOfScope event represent a variable go out of scope.
+        ```
+        GoOutOfScope {
+            ro: ResourceAccessPoint // must be a variable
+        },
+        ```
+        User case:
+        ```
+        fn main() { 
+            let x = 5; 
+            let y = x; // x and y both go out of scope
+        } 
+        ```
+    - InitRefParam <a name="InitRefParam"></a>
+    The InitRefParam event represent initialization of the parameters within a function
+        ```
+        InitRefParam {
+            param: ResourceAccessPoint, // the parameter in function
+        }
+        ```
+        User case:
+        ```
+        fn takes_ownership(some_string: String) { // initialize some_string
+            println!("{}", some_string) 
+        } 
+        ```
+## Modules
+1. [mdbook_plugin](mdbook_plugin)
```

Comment:
high level description of how the different components form the higher level component